### PR TITLE
allow to choose between copying, moving or linking the new files (hard and symlinks)

### PR DESCRIPTION
**Pull request by rotsix**
See #17 for more informations.

### CODE REVIEW

File: sort/fs_sort.go [hunk 1]
```diff
@@ -314,15 +341,15 @@ func (fs *fsSort) verbf(f string, args ...interface{}) {
...
+	switch fs.Action {
+	case moveAction:
+		err = move(src, dst)
+	case copyAction:
+		err = copy(src, dst)
+	case linkAction:
+		err = link(src, dst, fs.linkType)
 	}
+
 	return
```
Comment (jpillora) [inline sort/fs_sort.go:354]: `return errors.New("unknown action")`

File: main.go [hunk 2]
```diff
@@ -40,7 +40,8 @@ func main() {
 		FileLimit:         1000,
 		MinFileSize:       sizestr.Bytes(sizestr.MustParse("25MB")),
 		WatchDelay:        3 * time.Second,
-		AccuracyThreshold: 95, //100 is perfect match
+		AccuracyThreshold: 95, //100 is perfect match,
+		Action:            "move",
```
Comment (jpillora) [inline main.go:44]: `moveAction`

**Comment by jpillora** [comment]
Oh and can you also re-run `go run main.go --help` and paste it into the README.md?

File: main.go (hunk 2 above)
Comment (rotsix) [reply main.go:44]: Nop. `moveAction` is not defined in `main.go` as not an exported variable.

**Comment by jpillora** [comment]
Ah damn, I just realised we shouldn't change the existing API

https://godoc.org/github.com/jpillora/media-sort/sort#Config

Other apps might be using `Hardlink`, easy fix here is just to default to symlink. And also in future, they might want to set `Action` so we'll need to expose the enum types/consts

**Comment by jpillora** [comment]
Lastly, to maintain backward compatibility, `Hardlink: true` and `Action: move` (default) should automatically switch to `link`

**Comment by rotsix** [comment]
> `Action: move` (default) should automatically switch to `link`

Yes, so the default behavior should be hardlinking right?

**Comment by rotsix** [comment]
From what I understand, to maintain backward compatibility:

- `--action` defaults to `move`
- `--hardlink` forces to `link`, bypassing the `--action` flag

This way, existing apps can use the same options.

**Comment by jpillora** [comment]
Yep that's right, you could also error if the user tries to set hardlink,
and action to move/copy

**Comment by jpillora** [comment]
Cool looks good to me

**Comment by rotsix** [comment]
Thanks mate. That's a great tool, can't wait to use it to arrange my library.

**Comment by rotsix** [comment]
Note for later, Go standard library is also available on Windows.
I rewrote the `move` function to use pure stdlib and no extern calls, thus making the `move`, `link` and `copy` functions available on Windows too.
As I don't have any Windows machine to test it, I won't regroup them, but that's worth a try.

**Pull request merged**